Description: Fix build with SVT-AV1 4.0
 The SvtAv1PredStructure enum was removed from the public API in
 SVT-AV1 4.0.0. Use the raw integer value (2 = RANDOM_ACCESS) when
 building against SVT-AV1 >= 4.0.1.
Bug: https://gitlab.com/AOMediaCodec/SVT-AV1/-/issues/2341

--- a/libheif/plugins/encoder_svt.cc
+++ b/libheif/plugins/encoder_svt.cc
@@ -848,7 +848,11 @@ heif_error svt_start_sequence_encoding_intern(void* encoder_raw,
         //svt_config.pred_structure = SvtAv1PredStructure::SVT_AV1_PRED_LOW_DELAY_B;
         break;
       case heif_sequence_gop_structure_unrestricted:
+#if SVT_AV1_CHECK_VERSION(4, 0, 0)
+        svt_config.pred_structure = 2; // RANDOM_ACCESS
+#else
         svt_config.pred_structure = SvtAv1PredStructure::SVT_AV1_PRED_RANDOM_ACCESS;
+#endif
         break;
     }

